fondant/schema.py

```diff
@@ -2,56 +2,148 @@
 and pipelines.
 """
 
-import enum
 import typing as t
 
 import pyarrow as pa
 
-KubeflowCommandArguments = t.List[t.Union[str, t.Dict[str, str]]]
-
+from fondant.exceptions import InvalidTypeSchema
 
-class Type(enum.Enum):
-    """Supported types.
+KubeflowCommandArguments = t.List[t.Union[str, t.Dict[str, str]]]
 
-    Based on:
-    - https://arrow.apache.org/docs/python/api/datatypes.html#api-types
-    - https://pola-rs.github.io/polars/py-polars/html/reference/datatypes.html
+"""
+Types based on:
+- https://arrow.apache.org/docs/python/api/datatypes.html#api-types
+- https://pola-rs.github.io/polars/py-polars/html/reference/datatypes.html
+"""
+_TYPES: t.Dict[str, pa.DataType] = {
+    "null": pa.null(),
+    "bool": pa.bool_(),
+    "int8": pa.int8(),
+    "int16": pa.int16(),
+    "int32": pa.int32(),
+    "int64": pa.int64(),
+    "uint8": pa.uint8(),
+    "uint16": pa.uint16(),
+    "uint32": pa.uint32(),
+    "uint64": pa.uint64(),
+    "float16": pa.float16(),
+    "float32": pa.float32(),
+    "float64": pa.float64(),
+    "decimal128": pa.decimal128(38),
+    "time32": pa.time32("s"),
+    "time64": pa.time64("us"),
+    "timestamp": pa.timestamp("us"),
+    "date32": pa.date32(),
+    "date64": pa.date64(),
+    "duration": pa.duration("us"),
+    "string": pa.string(),
+    "utf8": pa.utf8(),
+    "binary": pa.binary(),
+    "large_binary": pa.large_binary(),
+    "large_utf8": pa.large_utf8(),
+}
+
+
+class Type:
+    """
+    The `Type` class provides a way to define and validate data types for various purposes. It
+      supports different data types including primitive types and complex types like lists.
     """
 
-    bool = pa.bool_()
-
-    int8 = pa.int8()
-    int16 = pa.int16()
-    int32 = pa.int32()
-    int64 = pa.int64()
-
-    uint8 = pa.uint8()
-    uint16 = pa.uint16()
-    uint32 = pa.uint32()
-    uint64 = pa.uint64()
-
-    float16 = pa.float16()
-    float32 = pa.float32()
-    float64 = pa.float64()
-
-    decimal = pa.decimal128(38)
-
-    time32 = pa.time32("s")
-    time64 = pa.time64("us")
-    timestamp = pa.timestamp("us")
-
-    date32 = pa.date32()
-    date64 = pa.date64()
-    duration = pa.duration("us")
-
-    string = pa.string()
-    utf8 = pa.utf8()
-
-    binary = pa.binary()
-
-    int8_list = pa.list_(pa.int8())
-
-    float32_list = pa.list_(pa.float32())
+    def __init__(self, data_type: t.Union[str, pa.DataType]):
+        self.value = self._validate_data_type(data_type)
+
+    @staticmethod
+    def _validate_data_type(data_type: t.Union[str, pa.DataType]) -> pa.DataType:
+        """
+        Validates the provided data type and returns the corresponding data type object.
+
+        Args:
+            data_type: The data type to validate.
+
+        Returns:
+            The validated `pa.DataType` object.
+        """
+        if isinstance(data_type, str):
```

Comment:
```suggestion
        if not isinstance(data_type, Type):
```

I think this is a bit more robust. What if I pass in something that is not a `Type` nor a string? :)